CsharpExtrasTest/Extensions/ArrayExtensionTest.cs

```diff
@@ -353,6 +353,81 @@ public void TestGivenArrayWithLargeNumberOfElementsWhenCheckForDuplicatesThenChe
             Assert.AreEqual(expectedResult, result, "Folded value should equal expected value");
         }
 
+        [Test]
+        [TestCaseSource("ProviderForFindFirstOccurrenceOfSet")]
+        public void Given_Array_When_FindFirstOccurrenceOfSet_Then_CorrectValueReturned(int[] arr,ISet<int> set,int expectedIndex, int expectedElement)
+        {
+            //Act
+            var actualResult = ArrayExtension.FindFirstOccurrenceOfSet(arr, set, 0, arr.Length-1);
+
+            //Assert
+            Assert.AreEqual(expectedIndex, actualResult.index);
+            Assert.AreEqual(expectedElement, actualResult.element);
+        }
+
+        [Test]
+        [TestCaseSource("ProviderForTo2DArray")]
+        public void Given_Array_When_To2DArray_Then_CorrectValueReturned(int[] array1D, int[,] expectedArray, ArrayOrientationClass.ArrayOrientation arrayOrientation)
+        {
+            //Act
+            int[,] actualArray = ArrayExtension.To2DArray(array1D, arrayOrientation);
+
+            //Assert
+            Assert.AreEqual(expectedArray, actualArray);
+        }
+
+        [Test]
+        [TestCaseSource("ProviderForDeepCopy")]
+        public void Given_Array_When_DeepCopy_Then_CorrectValueReturned(int[] expectedArray)
+        {
+            //Act
+            int[] actualArray = ArrayExtension.DeepCopy(expectedArray);
+
+            //Assert
+            Assert.AreEqual(expectedArray, actualArray);
+        }
+
+        [Test]
+        [TestCaseSource("ProviderForInverse")]
+        public void Given_Array_When_Inverse_Then_CorrectValueReturned(int[] array, IDictionary<int, IList<int>> expected)
+        {
+            //Act
+            IDictionary<int, IList<int>> actual = ArrayExtension.Inverse(array);
+
+            //Assert
+            Assert.AreEqual(expected, actual);
+        }
+
+        [Test]
+        public void Given_ThreeArraysOfStrings_When_ZipArray_Then_CorrectValueReturned()

```

Comment:
I would try to add test cases to validate the behaviour on arrays of different sizes. What happens in that case? Is an exception thrown? Is the smallest array size used?